src/main/java/org/kohsuke/github/GHPullRequest.java

```diff
@@ -557,6 +557,42 @@ public GHPullRequestReviewComment createReviewComment(String body, String sha, S
                 .wrapUp(this);
     }
 
+    /**
+     * Create review comment gh pull request review comment.
+     *
+     * @param body
+     *            the body
+     * @param sha
+     *            the sha
+     * @param path
+     *            the path
+     * @param startLine
+     *            For a single line comment, the line of the blob in the pull request diff that the comment applies to.
+     *            For a multi-line comment, the first line in the pull request diff that the comment applies to.
+     * @param endLine
+     *            For a multi-line comment, the last line of the range that the comment applies to. Otherwise, keep it
+     *            null.
+     * @return the gh pull request review comment
+     * @throws IOException
+     *             the io exception
+     */
+    public GHPullRequestReviewComment createReviewComment(String body,
+            String sha,
+            String path,
+            Integer startLine,
+            Integer endLine) throws IOException {
+        return root().createRequest()
+                .method("POST")
+                .with("body", body)
+                .with("commit_id", sha)
+                .with("path", path)
+                .with("start_line", endLine != null ? startLine : null)
+                .with("line", endLine != null ? endLine : startLine)
+                .withUrlPath(getApiRoute() + COMMENTS_ACTION)
```

Comment:
Wow, the doc page for this endpoint is a nightmare of "required but in some cases not required" statements. 😭

Reply:
Yes, it's not so easy to understand (and I can say the Gitlab API for multiline comments is really worse in terms of complexity than the Github one 😄). I tried to keep it as simple as possible in this client.